go/vt/mysqlctl/mysqld.go

```diff
@@ -635,6 +642,7 @@ func execCmd(name string, args, env []string, dir string, input io.Reader) (cmd
 // binaryPath does a limited path lookup for a command,
 // searching only within sbin and bin in the given root.
 func binaryPath(root, binary string) (string, error) {
+	noSocketFile()
```

Comment:
Anywhere we're going to check if a certain binary is available, implies we need to be running inside `mysqlctl`. This ensures we error out hard when we try to end up calling this somewhere else.

Everything is correct here today, this is a safety mechanism so that we don't accidentally add broken usage in the future.